frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelineRuns.cy.ts

```diff
@@ -111,251 +118,372 @@ const mockJobs = [
+  }),
+  buildMockRunKF({
+    display_name: 'Test archived run 2',
+    run_id: 'archived-run-2',
+    pipeline_version_reference: {
+      pipeline_id: pipelineId,
+      pipeline_version_id: 'test-version-1',
+    },
+    experiment_id: 'test-experiment-1',
+    created_at: '2024-02-05T00:00:00Z',
+    state: RuntimeStateKF.SUCCEEDED,
+  }),
+];
+
 describe('Pipeline runs', () => {
   beforeEach(() => {
     initIntercepts();
   });
 
-  describe('active runs', () => {
-    beforeEach(() => {
+  describe('Active runs', () => {
+    it('shows empty state', () => {
+      activeRunsTable.mockGetActiveRuns([]);
       pipelineRunsGlobal.visit(projectName, 'active');
+      activeRunsTable.findEmptyState().should('exist');
     });
 
-    it('renders the page with table data', () => {
-      activeRunsTable.findRowByName('Test active run 1');
-    });
-
-    describe('Table filter', () => {
-      it('filter by run name', () => {
-        // Verify initial run rows exist
-        activeRunsTable.findRows().should('have.length', 3);
-
-        // Select the "Name" filter, enter a value to filter by
-        pipelineRunsGlobal
-          .findActiveRunsToolbar()
-          .within(() => pipelineRunsGlobal.selectFilterByName('Name'));
-        pipelineRunsGlobal
-          .findActiveRunsToolbar()
-          .within(() => pipelineRunFilterBar.findNameInput().type('run 1'));
-
-        // Mock runs (filtered by typed run name)
-        activeRunsTable.mockGetRuns(
-          mockRuns.filter((mockRun) => mockRun.display_name.includes('run 1')),
-          1,
-        );
+    describe('with data', () => {
+      beforeEach(() => {
+        activeRunsTable.mockGetActiveRuns(mockActiveRuns);
+        pipelineRunsGlobal.visit(projectName, 'active');
+      });
 
-        // Verify only rows with the typed run name exist
-        activeRunsTable.findRows().should('have.length', 1);
+      it('renders the page with table data', () => {
         activeRunsTable.findRowByName('Test active run 1');
       });
 
-      it('filter by experiment', () => {
-        // Mock initial list of experiments
-        pipelineRunFilterBar.mockExperiments(mockExperiments);
+      it('archive a single run', () => {
+        const [runToArchive] = mockActiveRuns;
 
-        // Verify initial run rows exist
-        activeRunsTable.findRows().should('have.length', 3);
+        activeRunsTable.mockArchiveRun(runToArchive.run_id);
+        activeRunsTable.selectRowActionByName(runToArchive.display_name, 'Archive');
 
-        // Select the "Experiment" filter, enter a value to filter by
-        pipelineRunsGlobal
-          .findActiveRunsToolbar()
-          .within(() => pipelineRunsGlobal.selectFilterByName('Experiment'));
-        pipelineRunsGlobal
-          .findActiveRunsToolbar()
-          .within(() => pipelineRunFilterBar.findExperimentInput().type('Experiment 1'));
+        activeRunsTable.mockGetRuns([mockActiveRuns[1]], [runToArchive]).as('refreshAllRuns');
```

Comment:
why do you save `refreshAllRuns` but dont wait for it anywhere

Reply:
i would expect if you are saving it like this that you would wait after the submit to see that the request body is accurate to what was selected